lib/api/apiUtils/object/checkEncryption.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Checks the object encryption request against bucket encryption for matching
+ * SSE-S3 configuration
+ * @param {object} objectRequest - http request
+ * @param {object} serverSideEncryption - bucket encryption info
+ * @return {boolean} returns true if the object request has correct SSE-S3
+ * configuration
+ */
+function isValidSSES3(objectRequest, serverSideEncryption) {
+    // x-amz-server-side-encryption is allowed only if bucket
+    // encryption is enabled and if the value is AES256
+    // NOTE: object level encryption is not supported, but we allow
+    // encryption headers in the object request headers!
+    const sseHeader = objectRequest.headers['x-amz-server-side-encryption'];
+    const encryptionAlgorithm = 'AES256';
+
+    const result = ((!serverSideEncryption
```

Comment:
I think the first four lines of this conditional expression could be replaced by `(!serverSideEncryption && sseHeader) || ...`

Reply:
I had it like that..now it is just a bit more descriptive :)